File: atsa/src/main/java/jp/kaiz/atsassistmod/controller/trainprotection/TrainProtection.java
package jp.kaiz.atsassistmod.controller.trainprotection;

import cc.mirukuneko.realtrainmodrenewed.entity.TrainEntity;

public class TrainProtection {
    protected TrainEntity train;

    public void onTick(TrainEntity train, double distance) throws Exception {
        this.train = train;
    }

    public int getNotch(float speedH) {
        return 1;
    }

    public TrainProtectionType getType() {
        return TrainProtectionType.NONE;
    }

    public int getDisplaySpeed() {
        return Integer.MAX_VALUE;
    }
}

File: atsa/src/main/kotlin/jp/kaiz/atsassistmod/controller/trainprotection/TrainProtection.kt
// SPDX-License-Identifier: LGPL-3.0-or-later
// Copyright © 2026 mirukuneko and RealTrainModRenewed contributors
package jp.kaiz.atsassistmod.controller.trainprotection

import cc.mirukuneko.realtrainmodrenewed.entity.TrainEntity

open class TrainProtection {
    protected var train: TrainEntity? = null

    @Throws(Exception::class)
    open fun onTick(train: TrainEntity, distance: Double) {
        this.train = train
    }

    open fun getNotch(speedH: Float): Int = 1

    open fun getType(): TrainProtectionType = TrainProtectionType.NONE

    open fun getDisplaySpeed(): Int = Int.MAX_VALUE
}
